### ISSUE TITLE
Is the real cos similarity being used in Contextual Attention?

### ISSUE THREAD

**Issue by PangzeCheung** [JiahuiYu/generative_inpainting#441]
cos similarity need calculate norm of the patch. But in the code, I am not find where to calculate the norm of the xi.
yi = tf.nn.conv2d(xi, wi_normed, strides=[1,1,1,1], padding="SAME")
I am confuse about this question, thank you very much~

**Comment by JiahuiYu** [comment]
#281

**Comment by PangzeCheung** [comment]
Thank you very much~
